### ISSUE TITLE
showing images in grid layout 

### ISSUE THREAD

**Issue by ak2711** [amahi/android#156]
grid layout can be used to display images like this one

![12121](https://cloud.githubusercontent.com/assets/11472886/24320990/844fb9f2-1167-11e7-8139-f3240ce92981.PNG)

**Comment by cskc97** [comment]
This can be done using a recycler view with a grid layout manager with a span of 4

**Comment by cpg** [comment]
It's a nice idea. How to represent folders of images may also need some thought.

@ak2711 Please specify if you will be working on that or not.

**Comment by ak2711** [comment]
i would definitely like to work on it. folders can also be implemented  like the following
![screenshot_2017-03-26-02-12-39](https://cloud.githubusercontent.com/assets/11472886/24325952/da1fd9f4-11c9-11e7-9f90-963d50ec24fb.png)

**Comment by RanchoDbest** [comment]
@ak2711  Are u working on this issue??

**Comment by sd1998** [comment]
I would like to work on this issue

**Comment by cpg** [comment]
@sd1998 Please don't. Focus on the other issue you said you would be working on first #255

**Comment by mohitsvnit** [comment]
In "Welcome to amahi" server `shares/demo/photos` only contains images. Can it also contains folders?

**Comment by cpg** [comment]
potentially, yes

**Comment by cpg** [comment]
potentially, yes.

maybe do it in two parts. 1) start by assuming it does not. 2) later you can add that functionality if it makes sense.

**Comment by mohitsvnit** [comment]
OK. Because if folder exist then I have to create one separate fragment for that. First I will work on it assuming that folder does't exist. 
Can you tell which image format we are supporting?

**Comment by mohitsvnit** [comment]
@cpg and @csoni111  Should I create different fragment say `ServerFileImageFragment` for images having grid structure or should modify `ServerFileFragment` so that if images comes then it uses grid structure else use linear structure?  Please provide your valuable suggestion

**Comment by csoni111** [comment]
I think it would be better to modify `ServerFileFragment`. Duplicating the whole code for a single line change does not make sense.

**Comment by mohitsvnit** [comment]
@csoni111 I am thinking to change `Listview` into `RecyclerView`. What do you say?

**Comment by csoni111** [comment]
Yes do that but make sure you don't break any previous functionality.

**Comment by mohitsvnit** [comment]
@csoni111 functionality of ServerFileAdapter class is depends on multiple classes. Entire architecture seems really complex. I will try to complete it asap.
